crates/bevy_winit/src/lib.rs

```diff
@@ -340,6 +340,17 @@ fn handle_winit_event(
     #[cfg(feature = "trace")]
     let _span = bevy_utils::tracing::info_span!("winit event_handler").entered();
 
+    // Reinitialize system states if the world changed.
+    if !create_window.matches_world(app.world.id()) {
+        *create_window = SystemState::new(&mut app.world);
+    }
+    if !event_writer_system_state.matches_world(app.world.id()) {
+        *event_writer_system_state = SystemState::new(&mut app.world);
+    }
+    if !focused_windows_state.matches_world(app.world.id()) {
+        *focused_windows_state = SystemState::new(&mut app.world);
```

Comment:
This is potentially very expensive to be running repeatedly due to the archetype scan each time it's reinitialized, and may play poorly if queries were ever made into entities, as required by relations. This also happens in the winit event loop and blocks the main thread. I'm not sure this is something we want to be supporting.